File: C++/find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int szH = haystack.size(), szN = needle.size();
        
        for (int i = 0; i < (szH - szN + 1); i++) {
            int j = 0;
            for (j = 0; j < szN; j++) {
                if (haystack[i+j] != needle[j]) {
                    break;
                }
            }
            if (j == szN) {
                return i;
            }                
        }
        
        return -1;
    }
};

File: Kotlin/find-the-index-of-the-first-occurrence-in-a-string.kt
class Solution {
    fun strStr(haystack: String, needle: String): Int {
        val n = needle.length
        val m = haystack.length - n

        for (i in 0..m) {
            var found = true
            for (j in 0..n-1) {
                if (haystack[i+j] != needle[j]) {
                    found = false
                    break;
                }
            }
            if (found) {
                return i
            }
        }
        
        return -1
    }
}